Add pricing calculator to pricing page

kimberleyamackenzie opened the pull request:
## Description
Add a pricing calculator to the pricing page - this started as a hackathon project, and @meagancojocar has reviewed and also run past the Sales team.  Allows inputs for resource count, utilization, and deployment minutes and shows outputs of costs for monthly and daily usage.  Also adds a quick button to the calculator at the top of the page.  See some screenshots below for responsiveness.
<img width="670" alt="Screenshot 2023-11-27 at 2 48 24 PM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/18e6fb49-69f2-4b05-9964-f8d3d51ba930">
<img width="496" alt="Screenshot 2023-11-27 at 2 48 43 PM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/53c879b0-24c1-455a-aebb-fde782dfc524">
<img width="504" alt="Screenshot 2023-11-27 at 2 48 56 PM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/500f2796-787d-4ea9-ae83-09d19d4ef455">
<img width="1336" alt="Screenshot 2023-11-27 at 2 50 54 PM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/2af5c888-634b-48e7-a8a8-23a54fc393c7">
<img width="1313" alt="Screenshot 2023-11-27 at 2 51 09 PM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/2a708b46-370d-4204-9aad-e3955a32497c">
<img width="617" alt="Screenshot 2023-11-27 at 2 51 20 PM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/1033a5c8-3d0e-41ba-865a-8b95b3b1944c">
<img width="1261" alt="Screenshot 2023-11-27 at 2 54 44 PM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/49a7778e-dba8-43eb-8255-2bd48649d12d">


## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

meagancojocar commented:
We decided to ship this! Some changes to make:

- $20k per year cap, which is $1.7k a month, which is 5000 resources. When this amount is reached we want to change the $ amount to say "Contact sales for bulk discounts".
- Remove Enterprise. The calculator will just be for Team customers only.

kimberleyamackenzie commented on themes/default/theme/stencil/src/components/pulumi-pricing-calculator/pulumi-pricing-calculator.scss:
```diff
@@ -0,0 +1,349 @@
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    width: 140px;
+    // margin: auto;
```
```suggestion
```

zchase reviewed (changes requested):
We will want this run through our A/B testing tool. I'll make some time tomorrow to set it up.

cnunciato commented on themes/default/theme/stencil/src/components/pulumi-pricing-calculator/pulumi-pricing-calculator.scss:
```diff
@@ -0,0 +1,348 @@
+@import "../../../../src/scss/colors";
+.calculator {
+    margin-right: 36px;
+    margin-left: 36px;
```
Would strongly prefer Tailwind classes for these.

kimberleyamackenzie replied:
i based this off other stencil components where we have custom styles + i didn't see any usage of tailwind in any of those cases, so i made the assumption that it was intentional/tailwind + stencil didn't play well together.  but if that's not the case i can go back and tailwind where possible.

zchase reviewed (approved):
All good for the A/B test!

meagancojocar reviewed (approved):
Good to merge!

kimberleyamackenzie commented:
Cleaned up a couple items after syncing with @anita-trimbur - see screenshots below.  @meagancojocar I also found an error in math where I was not properly calculating the deployment minutes per month, so I fixed that too.

- [ ] Fixed bad math 😅 
- [ ] Decreased spacing next to calculator link at top
- [ ] Increased spacing below calculator header
- [ ] Gave the outputs a little more breathing room at below 1280
- [ ] Added "estimated deployment minutes" item in outputs to make it clear that the calculator was still reacting when that input was changed but remained below the free minutes
- [ ] Switched up the treatment for contact us to include link to sales form + blurred output

<img width="596" alt="Screenshot 2023-11-28 at 11 52 49 AM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/203659d5-ee4c-439c-9160-070dc4013b0c">
<img width="590" alt="Screenshot 2023-11-28 at 11 57 07 AM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/ddaa3c10-2589-4c50-84b4-76afeeadf529">


<img width="826" alt="Screenshot 2023-11-28 at 11 52 42 AM" src="https://github.com/pulumi/pulumi-hugo/assets/25756367/350b9d41-8a33-4f05-8a44-d8a9c0d7fd26">
